From 2e2926c40be9fd4f0dd0740ac73f3cf3ca683bf0 Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Fri, 20 Jun 2014 17:13:51 +0800
Subject: [PATCH 4374/5965] PD #83628: PMU driver update for AML1218, version
 0.70

1. update settings for register 0x3b/0x44/0x11b/0x11f by GAT_VBAT signal
   according David Wang
---
 arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd | 2 +-
 arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd   | 2 +-
 drivers/amlogic/power/aml_pmu/aml1218-sply.c    | 5 +++++
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
index e89489a3ed58..edcfa69bea5b 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
@@ -835,7 +835,7 @@ void root_func(){
         compatible = "amlogic, aml_pmu_prober";                 /* fixed for pmu driver, don't change   */
         #address-cells = <1>;
         #size-cells = <0>;
-        driver_version = "v0.60";
+        driver_version = "v0.70";
 //$$ DEVICE = "Pmu_axp202"
 //$$ L2 PROP_STR = "sub_type"
 //$$ L2 PROP_STR = "i2c_bus"
diff --git a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
index 2f99fb74cba2..f1041c44dc0c 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m102_1G.dtd
@@ -814,7 +814,7 @@ void root_func(){
         compatible = "amlogic, aml_pmu_prober";                 /* fixed for pmu driver, don't change   */
         #address-cells = <1>;
         #size-cells = <0>;
-        driver_version = "v0.60";
+        driver_version = "v0.70";
 //$$ DEVICE = "Pmu_axp202"
 //$$ L2 PROP_STR = "sub_type"
 //$$ L2 PROP_STR = "i2c_bus"
diff --git a/drivers/amlogic/power/aml_pmu/aml1218-sply.c b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
index 8850ac264402..70acdb24c975 100755
--- a/drivers/amlogic/power/aml_pmu/aml1218-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1218-sply.c
@@ -1320,6 +1320,11 @@ static int aml1218_update_state(struct aml_charger *charger)
     /*
      * limit duty cycle of DC3 according CHG_GAT_BAT_LV bit
      */
+    aml1218_set_bits(0x003b, (chg_status & 0x02000000) ? 0x00 : 0x40, 0x40);
+    aml1218_set_bits(0x0044, (chg_status & 0x02000000) ? 0x00 : 0x40, 0x40);
+    aml1218_write(0x011d, (chg_status & 0x02000000) ? 0x02 : 0x00);
+    aml1218_write(0x011f, (chg_status & 0x02000000) ? 0x02 : 0x00);
+
     aml1218_set_bits(0x0035, (chg_status & 0x02000000) ? 0x00 : 0x04, 0x07);
     aml1218_set_bits(0x003e, (chg_status & 0x02000000) ? 0x00 : 0x04, 0x07);
     aml1218_set_bits(0x0047, (chg_status & 0x02000000) ? 0x00 : 0x04, 0x07);
-- 
2.19.0

